commit 118e2d362a8dd0c1e6e4f5c4694ba193f77092ce
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Oct 3 21:31:48 2012 +0100

    Allow the number of cycles delay between sdcard peripheral writes to be specified on command line with sdhci-bcm2708.cycle_delay

Index: linux-3.2.46/drivers/mmc/host/sdhci-bcm2708.c
===================================================================
--- linux-3.2.46.orig/drivers/mmc/host/sdhci-bcm2708.c	2013-07-26 19:35:57.000000000 +0000
+++ linux-3.2.46/drivers/mmc/host/sdhci-bcm2708.c	2013-07-26 19:36:48.000000000 +0000
@@ -82,6 +82,8 @@
 #define REG_EXRDFIFO_EN     0x80
 #define REG_EXRDFIFO_CFG    0x84
 
+int cycle_delay=2;
+
 /*****************************************************************************\
  *									     *
  * Debug								     *
@@ -254,7 +256,7 @@
 		/* host->clock is the clock freq in Hz */
 		static hptime_t last_write_hpt;
 		hptime_t now = hptime();
-		ns_2clk = 2000000000/host->clock;
+		ns_2clk = cycle_delay*1000000/(host->clock/1000);
 
 		if (now == last_write_hpt || now == last_write_hpt+1) {
 			 /* we can't guarantee any significant time has
@@ -1517,6 +1519,7 @@
 module_param(sync_after_dma, bool, 0444);
 module_param(missing_status, bool, 0444);
 module_param(enable_llm, bool, 0444);
+module_param(cycle_delay, int, 0444);
 
 MODULE_DESCRIPTION("Secure Digital Host Controller Interface platform driver");
 MODULE_AUTHOR("Broadcom <info@broadcom.com>");
